### ISSUE TITLE
"forecast_template" deprecated as of 2023.08, will be removed in 2024.03

### ISSUE THREAD

**Issue by PARitter** [briis/hass-weatherflow2mqtt#244]
### New Feature

Home assistant is changing how forecast data is retrieved from weather entities.  The "forecast_template" used by hass-weatherflow2mqtt is deprecated and will be removed as of the March, 2024 release.

See HASS release update: https://www.home-assistant.io/blog/2023/09/06/release-20239/#weather-forecast-service

There is a new service in HASS, "Weather: get forecast", to retrieve forecast data but this does not appear to be implemented in hass-weatherflow2mqtt.

If this is not addressed forecast data from weatherflow hub will not be available after updating to 2024.03.

**Comment by PARitter** [comment]
That certainly addresses the short-term impact of this change and suppresses the current warning.  But I presume (without knowledge) this remains deprecated and will be a breaking change in 2024.03.  When the "forecast_template" is finally removed, its component parts, "forecast_hourly_template" and "forecast_daily_template" will almost certainly be removed with it.

**Comment by drothenberger** [comment]
Those tags for the weather template are new and I believe were added as part of this new change, so I don't think they are going away. If they were going away, I would expect a warning for using them as well.

There is no Tempest integration here that can provide a weather entity and the corresponding get_weather service, so I think we will have to continue to use the weather template. 

TL;DR: I think this is the right long-term approach, but I'm no expert.

**Comment by drothenberger** [comment]
This is the commit in 2023.09 that added the new forecast_daily_template and forecast_hourly_template attributes to the weather template: https://github.com/home-assistant/core/commit/207e3f90a66e1b34a9dce05fc2caafc3662150d2

**Comment by mjmeli** [comment]
Yes @drothenberger is right, forecast_template is being deprecated because forecast_hourly_template and forecast_daily_template are replacing it.

There was a change in 2023.9.3 that exposed the erroneous keys in the log message. These are the ones that need to be removed:
`Only valid keys in Forecast are allowed, unallowed keys: ({'precip_icon', 'precip_type', 'wind_direction_cardinal', 'conditions'})`

**Comment by briis** [comment]
I have released 3.2.2 that closes this issue. Thank you @mjmeli
